datafusion/functions/src/string/overlay.rs

```diff
@@ -76,54 +80,107 @@ impl ScalarUDFImpl for OverlayFunc {
 
     fn invoke(&self, args: &[ColumnarValue]) -> Result<ColumnarValue> {
         match args[0].data_type() {
-            DataType::Utf8 => make_scalar_function(overlay::<i32>, vec![])(args),
+            DataType::Utf8View | DataType::Utf8 => {
+                make_scalar_function(overlay::<i32>, vec![])(args)
+            }
             DataType::LargeUtf8 => make_scalar_function(overlay::<i64>, vec![])(args),
             other => exec_err!("Unsupported data type {other:?} for function overlay"),
         }
     }
 }
 
+macro_rules! process_overlay {
```

Comment:
It would be nice to have some sort of trait in arrow-rs that allowed us to write this as a generic function

It actually does have 
https://github.com/apache/arrow-rs/blob/2461a16c19ee5032531b1c05dd7e7192bc842e0f/arrow-string/src/like.rs#L158-L161

But that is not public

@XiangpengHao do you know of anything that is pub?

We could also implement such a trait for DataFusion's convenience, and then propose upstreaming it 🤔

Reply:
I was just behind -- it seems that @Omega359  did exactly this in `StringArrayType` https://github.com/apache/datafusion/pull/11941

https://github.com/apache/datafusion/blob/94034bef9357b166d5593de90b00a35563eec3f8/datafusion/functions/src/unicode/lpad.rs#L251-L263

Maybe we can start to pull that trait into its own module and start reusing it across the string functions 🤔 

Also, there is the `ArrayAccessor` pattern used elegantly by @devanbenz in https://github.com/apache/datafusion/pull/11967 🤔